exasol/slc_ci/lib/check_if_build_needed.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from pathlib import Path
+from typing import List
+
+from exasol.slc_ci.lib.branch_config import build_always
+from exasol.slc_ci.lib.get_build_config_model import get_build_config_model
+from exasol.slc_ci.lib.git_access import GitAccess
+from exasol.slc_ci.lib.github_access import GithubAccess
+
+
+def get_all_affected_files(git_access: GitAccess, base_branch: str) -> List[Path]:
+    base_last_commit_sha = git_access.get_head_commit_sha_of_branch(base_branch)
```

Comment:
This is not needed, because we get the ref from GitHub via the command line parameter

Reply:
As discussed on the phone, we still need it because in case of a PR, `base-ref` is a branch name